handlers/facebookapp/facebookapp.go

```diff
@@ -662,20 +662,34 @@ func (h *handler) DescribeURN(ctx context.Context, channel courier.Channel, urn
 	base, _ := url.Parse(graphURL)
 	path, _ := url.Parse(fmt.Sprintf("/%s", urn.Path()))
 	u := base.ResolveReference(path)
-
 	query := url.Values{}
-	query.Set("access_token", accessToken)
-	u.RawQuery = query.Encode()
-	req, _ := http.NewRequest(http.MethodGet, u.String(), nil)
-	rr, err := utils.MakeHTTPRequest(req)
-	if err != nil {
-		return nil, fmt.Errorf("unable to look up contact data:%s\n%s", err, rr.Response)
-	}
 
-	// read our name
-	name, _ := jsonparser.GetString(rr.Body, "name")
+	if fmt.Sprint(channel.ChannelType()) == "FBA" {
+		query.Set("fields", "first_name,last_name")
+		query.Set("access_token", accessToken)
 
-	return map[string]string{"name": name}, nil
+		u.RawQuery = query.Encode()
+		req, _ := http.NewRequest(http.MethodGet, u.String(), nil)
+		rr, err := utils.MakeHTTPRequest(req)
+		if err != nil {
+			return nil, fmt.Errorf("unable to look up contact data:%s\n%s", err, rr.Response)
+		}
```

Comment:
I think it would be better to avoid repeating these lines below

so first if block to set the fields and one last if block to make the name from the response then always have the same last line as `return map[string]string{"name": name}, nil`